src/sentry/migrations/0506_slugify_invalid_monitors.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 2.2.28 on 2023-07-07 15:51
+import re
+
+from django.db import migrations
+from django.utils.text import slugify
+
+from sentry.monitors.models import MAX_SLUG_LENGTH
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+
+    slug_regex = re.compile("^[a-zA-Z\\d_-]+")
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.filter()):
+        # Nothing to migrate if the monitor already has a valid slug
+        monitor_slug = monitor.slug
+        if slug_regex.match(monitor_slug):
+            continue
```

Comment:
Does it make sense to just attempt to slugify and write if it changes?

Reply:
fair